pkg/client/client_v2.go

```diff
@@ -528,10 +540,15 @@ func (c *clientV2) findUnit(id string, unitType UnitType) *Unit {
 	return nil
 }
 
-// unitChanged triggers the send goroutine to send a new observed state
-func (c *clientV2) unitChanged() {
-	if len(c.kickSendObservedCh) <= 0 {
-		c.kickSendObservedCh <- struct{}{}
+// unitsStateChanged notifies checkinWriter that there is new
+// state data to send.
+func (c *clientV2) unitsStateChanged() {
```

Comment:
I prefer the old name `unitChange`, since the unit itself contains a `State` field and I believe we want to call this when anything in the unit changes not just the state.

Reply:
Fwiw this is only called in two places:
- In `syncUnits`, when a unit is removed
- In `(*Unit).UpdateState`, when a new `UnitState` is received. The fields that can be modified when this happens are `state`, `stateMsg`, and `statePayload`.

So I think in this case "state" is fairly accurate -- most changes to the `Unit` structure itself will not trigger this reload (because the state is the only part that needs to be reported back to Agent). But if you still prefer the other name after this explanation I can change it :-)

Reply:
Re-reading the code I think its fine as is, thanks.